docs/aws/audit/kmsmonitoring/rules/secret_encrypted_kms_cmk.mdx

```diff
@@ -23,6 +23,281 @@ ISO27001, HIPAA, NISTCSF, PCIDSS
+   If you have an existing secret that is not encrypted with a CMK, you can update it to use the CMK.
+
+   ```sh
+   aws secretsmanager update-secret --secret-id <Your-Secret-ID> --kms-key-id <Your-CMK-Key-ID-or-ARN>
+   ```
+
+By following these steps, you ensure that your secrets in AWS Secret Manager are encrypted using Customer Master Keys (CMKs) in AWS KMS, enhancing the security of your sensitive information.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that Secret Manager secrets are encrypted with Customer Managed Keys (CMKs) in KMS using Python scripts, you can follow these steps for AWS, Azure, and GCP:
+
+### AWS (Amazon Web Services)
+
+1. **Install AWS SDK for Python (Boto3):**
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a KMS Key:**
+   ```python
+   import boto3
+
+   kms_client = boto3.client('kms')
+   response = kms_client.create_key(
+       Description='My CMK for Secret Manager',
+       KeyUsage='ENCRYPT_DECRYPT',
+       Origin='AWS_KMS'
+   )
+   cmk_id = response['KeyMetadata']['KeyId']
+   print(f"Created CMK with ID: {cmk_id}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   import boto3
+
+   secrets_client = boto3.client('secretsmanager')
+   secret_name = "my_secret"
+   secret_value = "my_secret_value"
+
+   response = secrets_client.create_secret(
+       Name=secret_name,
+       SecretString=secret_value,
+       KmsKeyId=cmk_id
+   )
+   print(f"Created secret with ARN: {response['ARN']}")
+   ```
+
+### Azure (Microsoft Azure)
+
+1. **Install Azure SDK for Python:**
+   ```bash
+   pip install azure-identity azure-keyvault-secrets azure-keyvault-keys
+   ```
+
+2. **Create a Key Vault and a Key:**
+   ```python
+   from azure.identity import DefaultAzureCredential
+   from azure.keyvault.keys import KeyClient
+   from azure.keyvault.secrets import SecretClient
+
+   credential = DefaultAzureCredential()
+   key_vault_name = "myKeyVault"
+   key_vault_uri = f"https://{key_vault_name}.vault.azure.net"
+
+   key_client = KeyClient(vault_url=key_vault_uri, credential=credential)
+   key_name = "myCMK"
+   key = key_client.create_key(key_name, "RSA")
+   print(f"Created key with name: {key.name}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   secret_client = SecretClient(vault_url=key_vault_uri, credential=credential)
+   secret_name = "mySecret"
+   secret_value = "mySecretValue"
+
+   secret = secret_client.set_secret(secret_name, secret_value, key_id=key.id)
+   print(f"Created secret with name: {secret.name}")
+   ```
+
+### GCP (Google Cloud Platform)
+
+1. **Install Google Cloud SDK for Python:**
+   ```bash
+   pip install google-cloud-secret-manager google-cloud-kms
+   ```
+
+2. **Create a KMS Key:**
+   ```python
+   from google.cloud import kms_v1
+
+   client = kms_v1.KeyManagementServiceClient()
+   project_id = "my-project"
+   location_id = "global"
+   key_ring_id = "my-key-ring"
+   key_id = "my-key"
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>